### DISCUSSION TITLE
Heat ports in component Buildings.Fluid.FixedResistances.BaseClasses.Pipe

### DISCUSSION THREAD

**Discussion by JordiGarciaCespedes** [lbl-srg/modelica-buildings#3647] [Ideas]
Dear all,
Concerning the component `Buildings.Fluid.FixedResistances.BaseClasses.Pipe`, in the Documentation Browser it is said that _this model can be used for modeling the heat exchange between the pipe and the environment_. However there is no heat port to do so in the uppermost layer of the model. For this reason I modify this particular model to inherite correctly the heat ports of each of the mixing volumes for each segment in which the pipe is discretized. I inserted the component `Modelica.Fluid.Interfaces.HeatPorts_a` and made a connection of each of the heat ports of the mixing volumes with this component, within the code. In the `equation` block, instead of using the current code:
```
  connect(port_a, preDro.port_a);
  connect(preDro.port_b, vol[1].ports[1]);
  if nSeg > 1 then
    for i in 1:(nSeg - 1) loop
      connect(vol[i].ports[2], vol[i + 1].ports[1]);
    end for;
  end if;
  connect(vol[nSeg].ports[2], port_b)
```
I use this:
```
connect(port_a, preDro.port_a) ;
 connect(preDro.port_b, vol[1].ports[1]);
 connect(vol[1].heatPort,heatPorts_a[1]);
  if nSeg > 1 then
    for i in 1:(nSeg - 1) loop
      connect(vol[i].ports[2], vol[i + 1].ports[1]);
    end for;
    for i in 2:nSeg loop
      connect(vol[i].heatPort,heatPorts_a[i]);
      end for;
  end if;
  connect(vol[nSeg].ports[2], port_b)
```
So now I can use the heat port of the pipe to connect it to the ground.
Is this ok? Thanks for checking. If correct, I hope this might be useful to modify this component in future versions of the library.

p.s.:  Sorry if I do not use proper programming terms.

**Comment by JayHuLBL** [comment]
@JordiGC-UPC Did you look into the model `Buildings.Fluid.FixedResistances.Pipe` and see how it is used in `Buildings.Fluid.FixedResistances.Examples.Pipe`?

**Reply by JordiGarciaCespedes** [reply]
Dear Jianjun,
Yes, I saw that the component `Buildings.Fluid.FixedResistances.Pipe` is doing what I expose above, but offering the possibility of either collect all the mixing voume heatports in one or not. 
However, now I remember that when I face it for the first time I preferred to actuate on the BaseClass component to get a more customized modelt for my needs. For example, I prefer to specify externally and separatelly the thermal resistance due to the pipe wall material and the (optional) insulating layer.
Anyway, all is more clear now ;-)
Thanks!
